Build fails with JULIA_THREADS=1 again

Fails at f7d75b5; my last successful build was at d2f6428.

```
$ make
    JULIA usr/lib/julia/inference0.ji
LLVM ERROR: Program used external function 'jl_get_ptls_states.ptr' which could not be resolved!
make[1]: *** [/home/sachs/src/julia-multithreading/usr/lib/julia/inference0.ji] Error 1
make: *** [julia-inference] Error 2
```

--- a/src/codegen.cpp
+++ b/src/codegen.cpp
@@ -3660,7 +3660,18 @@ static void finalize_gc_frame(Function *F)
 
 #ifdef JULIA_ENABLE_THREADING
     if (imaging_mode) {
-        Value *getter = tbaa_decorate(tbaa_const, new LoadInst(jltls_states_func_ptr, "", ptlsStates));
+        Value *getter;
+        if (GlobalVariable *GV = M->getGlobalVariable(jltls_states_func_ptr->getName())) {
+            getter = tbaa_decorate(tbaa_const, new LoadInst(GV, "", ptlsStates));
+        }
+        else {
+            // If the global variable doesn't exist, we are coping code out of
+            // the shadow_module to run, in which case, we just use the
+            // function directly.
+            getter = ConstantExpr::getIntToPtr(
+                ConstantInt::get(T_size, (intptr_t)jl_get_ptls_states_getter()),
+                jltls_states_func->getFunctionType()->getPointerTo());
+        }
         ptlsStates->setCalledFunction(getter);
     }
     ptlsStates->setAttributes(jltls_states_func->getAttributes());

